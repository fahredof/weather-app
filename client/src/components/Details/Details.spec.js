import React from 'react';
import Details from "../Details";

describe(`Details of Weather`, () => {
    const mainCity = {
        wind: `Wind 3 m/s`,
        overcast: `overcast clouds`,
        pressure: `Pressure 1011 hpa`,
        humidity: `Humidity 66 %`,
        coordinates: `Coordinates [54.73, 55.95]`,
    };
    const mainCityError = {
        wind: ``,
        overcast: ``,
        pressure: ``,
        humidity: ``,
        coordinates: ``,
    };
    const props = {
        mainCity
    };

    describe(`отображение города`, () => {
        const defCity = shallow(<Details {...props}/>);
        it(`отображается корректно`, () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });

    describe('ошибка при получении города', () => {
        const nextProps = {
            ...props,
            mainCity: mainCityError
        };
        const defCity = shallow(<Details {...nextProps} />);
        it('отображается корректно', () => expect(shallowToJson(defCity)).toMatchSnapshot());
    });
});