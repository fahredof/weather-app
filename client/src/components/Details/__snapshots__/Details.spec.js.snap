// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details of Weather отображение города отображается корректно 1`] = `
<div
  className="det"
>
  <p>
    Wind 3 m/s
  </p>
  <p>
    overcast clouds
  </p>
  <p>
    Pressure 1011 hpa
  </p>
  <p>
    Humidity 66 %
  </p>
  <p>
    Coordinates [54.73, 55.95]
  </p>
</div>
`;

exports[`Details of Weather ошибка при получении города отображается корректно 1`] = `
<div
  className="det"
>
  <p />
  <p />
  <p />
  <p />
  <p />
</div>
`;
